keps/sig-api-machinery/0006-apply.md

```diff
@@ -443,6 +443,106 @@ y,Initialized,PodScheduled,Ready,/var/run/secrets/kubernetes.io/serviceaccount]}
 | 400 in base64 | \~ | \~ | \~ |
 | 300 in raw bytes | \~ | \~ | \~ |
 
+##### Generating String Tables
+
+A traversal across type declarations in the Kubernetes codebase will be used to
+gather all field names and enum values, which can be used as a starting point
+for the string table.
+
+The strings found in type declarations will be supplemented with other strings
+found to be frequently used in fieldsets. Commonly used string values and
+unstructured map keys are good candidates. For example,
+`/var/run/secrets/kubernetes.io/serviceaccount` was found by direct inspection
+as being a great string table entry; it is quite long, and it is frequently
+used. We expect to find more string table entries like this via tools and direct
+inspection, and will keep track of them in a separate list that will then be
+merged together with the strings from type declarations when generating string
+tables.
+
+A couple considerations when generating the string table:
+
+- Not all strings are worth adding to the string table. If the `!<base64>`
+  string is longer than the string it would replace, then it clearly makes no
+  sense to add it to the dictionary.
+- Since the base64 strings of low ints are shorter, they should be used for
+  more frequently used strings in fieldsets.
+
+Fine tuning the string table based on what strings are most frequent is
+something we can iterate on. If the initial string table size is small, ordering
+by frequency might not matter that much, but if/when the table gets larger,
+ensuring that frequently used strings are assigned low numbers in the table will
+become increasingly valuable.
+
+String table generation must be reproducable. The generator and all needed
```

Comment:
Are you sure? I was expecting to check in the result of this process and not do it very frequently at all.

I agree checking in the hand-curated stuff separately from the machine-generated stuff seems like a good idea.

Reply:
We can sort out exactly what we need in implementation. I'll drop this from the doc, I don't think it's really needed here.